src/err/mod.rs

```diff
@@ -153,7 +182,59 @@ impl PyErr {
     /// # });
     /// ```
     #[inline]
-    pub fn new<T, A>(args: A) -> PyErr
+    pub fn new_arg<T, A>(arg: A) -> PyErr
+    where
+        T: PyTypeInfo,
+        A: IntoPy<PyObject> + Send + Sync + 'static,
+    {
+        PyErr::from_state(PyErrState::Lazy(Box::new(move |py| {
+            PyErrStateLazyFnOutput {
+                ptype: T::type_object(py).into(),
+                pvalue: PyTuple::new(py, &[arg.into_py(py)]).into(),
```

Comment:
It seems unfortunate that we have to (understandably) wrap everything inside a `PyTuple` to make things behave as intended here.

I wonder if there's performance justification to have a special case for e.g. strings (I assume one-arg strings are 99% of the calls).

Reply:
I mean, sure, we can do a check and only wrap if it's a tuple...